components/script/dom/window.rs

```diff
@@ -1635,8 +1639,12 @@ impl Window {
     }
 
     pub fn suspend(&self) {
-        // Suspend timer events.
-        self.upcast::<GlobalScope>().suspend();
+        if let Some(document) = self.document.get() {
+            if !document.salvageable() {
```

Comment:
I think we will still want to suspend here. AFAICT if the document is not `salvageable` it will be discarded from the session history entirely.

Reply:
For a reason I haven't figured out yet, if a salvageable document is suspended, and resumed, this test still timeouts because the `setTimeout` callback is never called https://github.com/servo/servo/blob/1fe0f32059270e2d3e42b7c7184c2adf56f3f872/tests/wpt/web-platform-tests/html/browsers/browsing-the-web/unloading-documents/unload/006-1.html#L6

Reply:
Interesting... Something else may be wrong here then. I would still leave this how it was.

Reply:
I've digged a bit deeper, and this seems to be a problem at the level of `timers`, because the document is properly suspended and resumed, but once resumed the timer seems to be ignored because of it's id(see `OneShotTimer.fire_timer`). I'll be debugging it a bit more...

Reply:
@gterzian interesting, sounds like another followup. Let's open an issue for it and mark the test as failing.

Reply:
Filed https://github.com/servo/servo/issues/20663